apps/sms/js/notify.js

```diff
@@ -56,16 +73,22 @@
 
   var Notify = {
     ringtone: function notification_ringtone() {
-      if (settings['audio.volume.notification'] &&
-          settings['notification.ringtone']) {
-        ringtone(settings['notification.ringtone']);
-      }
+      return getSettings(
+        ['audio.volume.notification', 'notification.ringtone']
```

Comment:
actually we don't reuse them so much, we use them only when we receive a message and we're in the thread for this message (and maybe class0 messages too). That's why I choose to retrieve the setting only when we need it, because in these cases we don't mind some ms latency. I thought that adding an observer for these cases would lead to more code running than just retrieving when we need them.

What do you think?